src/workerd/api/gpu/gpu-device.c++

```diff
@@ -28,6 +29,49 @@ jsg::Ref<GPUBuffer> GPUDevice::createBuffer(jsg::Lock& js, GPUBufferDescriptor d
   return jsg::alloc<GPUBuffer>(js, kj::mv(buffer), kj::mv(desc), device_, kj::addRef(*async_));
 }
 
+jsg::Ref<GPUTexture> GPUDevice::createTexture(jsg::Lock& js, GPUTextureDescriptor descriptor) {
+  wgpu::TextureDescriptor desc{};
+  desc.label = descriptor.label.cStr();
+
+  KJ_SWITCH_ONEOF(descriptor.size) {
+    KJ_CASE_ONEOF(coords, jsg::Sequence<GPUIntegerCoordinate>) {
+      switch (coords.size()) {
```

Comment:
Non-blocking nit: Some comments here explaining the switch case here a bit would be worthwhile.